packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -307,7 +307,10 @@ See ${createDocumentationLink({
     );
 
     safelyRunOnBrowser(() => {
-      if (options.future?.preserveSharedStateOnUnmount === undefined) {
+      if (
+        __DEV__ &&
+        options.future?.preserveSharedStateOnUnmount === undefined
+      ) {
```

Comment:
I originally wanted to prevent this message from adding too much noise in server logs (ie Next), but I thought about it again now and we at least want them to notice the message at least once so that they explicitly set the flag. I'll remove it.